### PR TITLE
Docs: Update no-confusing-arrow with the new default option

### PR DESCRIPTION
**Pull request by zypA13510**
**What is the purpose of this pull request? (put an "X" next to item)**

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
- Removed redundant sentence from the summary as the default for allowParens is now true.
- Updated examples according to the new default.

**Is there anything you'd like reviewers to focus on?**
N/A

### CODE REVIEW

File: docs/rules/no-confusing-arrow.md [hunk 1]
```diff
@@ -32,6 +30,8 @@ Examples of **correct** code for this rule:
 /*eslint no-confusing-arrow: "error"*/
 /*eslint-env es6*/
 
+var x = (a) => 1 ? 2 : 3;
```
Comment (g-plane) [inline docs/rules/no-confusing-arrow.md]: `var x = (a) => 1 ? 2 : 3;` is still an incorrect case.

Comment (zypA13510) [reply docs/rules/no-confusing-arrow.md]: Do you squash merge? I mean, do I need to amend my commit then force push, or just add another commit?

Comment (g-plane) [reply docs/rules/no-confusing-arrow.md]: We use squash merge, so both amending the last commit and adding new commit are OK.

**Comment by g-plane** [comment]
And, please sign our CLA.

**Comment by zypA13510** [comment]
Done, also added a new example because it seems appropriate to me.

**Pull request merged**